packages/validator/src/spectral/spectral-validator.js

```diff
@@ -97,30 +98,40 @@ function convertResults(spectralResults, logger) {
 }
 
 /**
- * Creates a new spectral instance, sets up the ruleset, then returns the spectral instance.
+ * Creates a new Spectral instance, sets up the ruleset, then returns the spectral instance.
  *
- * @param {*} logger the logger object used to log messages
- * @param {string} rulesetFileOverride the path to a ruleset as given by an argument
- * @param {*} chalk an object used to colorize messages
- * @returns the spectral instance
+ * @param {*} opts an object containing the options
+ * @param {*} opts.logger the logger to use for logging messages
+ * @param {*} opts.chalk the chalk object for displaying messages
+ * @param {string} opts.rulesetFileOverride an optional ruleset filename
+ * @returns the new Spectral instance
  */
-async function setup(logger, rulesetFileOverride, chalk) {
+async function setup(opts) {
   const spectral = new Spectral();
 
-  // spectral only supports reading a config file in the working directory
-  // but we support looking up the file path for the nearest file (if one exists)
+  // Grab the context fields we need.
+  const logger = opts.logger;
+  let rulesetFileOverride = opts.rulesetFileOverride;
+  const chalk = opts.chalk;
```

Comment:
You could use a similar pattern as suggested above here, didn't think about that before. Something like
```js
const { chalk, logger } = opts;
let { rulesetFileOverride } = opts;
```

Reply:
done